test/e2e/pair-liquidity-info-history-v2-db.service.e2e-spec.ts

```diff
@@ -0,0 +1,220 @@
+  id: 333,
+  pairId: 2,
+  eventType: EventType.PairMint,
+  reserve0: new Decimal(1000),
+  reserve1: new Decimal(1000),
+  deltaReserve0: new Decimal(1000),
+  deltaReserve1: new Decimal(1000),
+  fiatPrice: new Decimal(0),
+  height: 300003,
+  microBlockHash: 'mh_entry3',
+  microBlockTime: 3000000000003n,
+  transactionHash: 'th_entry3',
+  transactionIndex: 300003n,
+  logIndex: 1,
+  createdAt: new Date(),
+  updatedAt: new Date(),
+};
+const historyEntry4: PairLiquidityInfoHistoryV2 = {
+  id: 444,
+  pairId: 2,
+  eventType: EventType.SwapTokens,
+  reserve0: new Decimal(1050),
+  reserve1: new Decimal(950),
+  deltaReserve0: new Decimal(50),
+  deltaReserve1: new Decimal(-50),
+  fiatPrice: new Decimal(0),
+  height: 300003,
+  microBlockHash: 'mh_entry4',
+  microBlockTime: 3000000000003n,
+  transactionHash: 'th_entry4',
+  transactionIndex: 300003n,
+  logIndex: 2,
+  createdAt: new Date(),
+  updatedAt: new Date(),
+};
+
+describe('PairLiquidityInfoHistoryV2DbService', () => {
+  let service: PairLiquidityInfoHistoryV2DbService;
+  let prismaService: PrismaService;
+
+  beforeAll(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [PairLiquidityInfoHistoryV2DbService, PrismaService],
+    }).compile();
+
+    service = module.get<PairLiquidityInfoHistoryV2DbService>(
+      PairLiquidityInfoHistoryV2DbService,
+    );
+    prismaService = module.get<PrismaService>(PrismaService);
+  });
+
+  beforeEach(async () => {
+    await prismaService.token.createMany({ data: [token1, token2, token3] });
+    await prismaService.pair.createMany({ data: [pair1, pair2, pair3] });
+    await prismaService.pairLiquidityInfoHistoryV2.createMany({
+      data: [historyEntry1, historyEntry2, historyEntry3, historyEntry4],
+    });
+  });
+
+  afterEach(async () => {
+    await prismaService.pairLiquidityInfoHistoryV2.deleteMany();
+    await prismaService.pair.deleteMany();
+    await prismaService.token.deleteMany();
+  });
+
+  afterAll(async () => {
+    await prismaService.$disconnect();
+  });
+
+  describe('upsert', () => {
+    it('should correctly upsert an existing entry', async () => {
+      const updatedEntry = {
+        pairId: historyEntry2.pairId,
+        eventType: EventType.PairBurn,
+        reserve0: new Decimal(500),
+        reserve1: new Decimal(500),
+        deltaReserve0: new Decimal(-500),
+        deltaReserve1: new Decimal(-500),
+        fiatPrice: new Decimal(0),
+        height: 200002,
+        microBlockHash: historyEntry2.microBlockHash,
+        microBlockTime: 2000000000002n,
+        transactionHash: historyEntry2.transactionHash,
+        transactionIndex: 200002n,
+        logIndex: historyEntry2.logIndex,
+      };
+      await service.upsert(updatedEntry);
+      const entry = await prismaService.pairLiquidityInfoHistoryV2.findFirst({
+        where: { id: historyEntry2.id },
+      });
+      expect(entry?.reserve0).toEqual(new Decimal(500));
```

Comment:
i guess we could also validate here if any of the value sare changed that should not be changed by doing a full data comparison, checkout snapshots for this :)